Index: likewise-open-4.1.2956/domainjoin/libdomainjoin/src/djdaemonmgr_nonmac.c
===================================================================
--- likewise-open-4.1.2956.orig/domainjoin/libdomainjoin/src/djdaemonmgr_nonmac.c	2008-08-04 01:01:17.000000000 -0500
+++ likewise-open-4.1.2956/domainjoin/libdomainjoin/src/djdaemonmgr_nonmac.c	2008-08-12 13:50:57.000000000 -0500
@@ -615,8 +615,13 @@
 
     }
 
+#if 0  /* Disabled to allow Ubuntu's upstart process and packaging 
+	  to manage the daemon's restart on boot */
+
     LW_TRY(exc, DJConfigureForDaemonRestart(pszName, bStatus, pszStartPriority, pszStopPriority, &LW_EXC));
 
+#endif
+
 cleanup:
     ;
 }
