diff --git a/domainjoin/libdomainjoin/src/djpamconf.c b/domainjoin/libdomainjoin/src/djpamconf.c
index c5cb73e..e8eb3e4 100644
--- a/domainjoin/libdomainjoin/src/djpamconf.c
+++ b/domainjoin/libdomainjoin/src/djpamconf.c
@@ -1736,7 +1736,8 @@ static void PamLwidentityEnable(const char *testPrefix, const DistroInfo *distro
 
         if(PamModuleAlwaysDeniesDomainLogins(phase, module) && (
                     !strcmp(control, "required") ||
-                    !strcmp(control, "requisite")))
+                    !strcmp(control, "requisite") ||
+		    !strcmp(control, "[success=1 default=ignore]")))
             break;
 
         if(PamModuleIsLwidentity(phase, module))
