--- export/doc/editrc.5.roff	2009-09-05 13:37:56.000000000 +0200
+++ ../doc/editrc.5.roff	2009-09-05 13:47:56.000000000 +0200
@@ -146,7 +146,7 @@
 .Ar command
 can contain control characters of the form
 .Sm off
-.Sq No ^ Ar character
+.Sq ^character
 .Sm on
 .Po
 e.g.
@@ -172,7 +172,7 @@
 .It Ic \ev
 Vertical tab
 .Sm off
-.It Sy \e Ar nnn
+.It Sy \ennn
 .Sm on
 The ASCII character corresponding to the octal number
 .Ar nnn .
@@ -207,11 +207,11 @@
 causing an error.
 .Fl v
 causes messages to be verbose.
-.It Ic edit Op Li on | Li off
+.It Ic edit Op on | off
 Enable or disable the
 .Nm editline
 functionality in a program.
-.It Ic history Ar list | Ar size Dv n | Ar unique Dv n
+.It Ic history Ar list | Ar size n | Ar unique n
 The
 .Ar list
 command lists all entries in the history.
